minesweeper.py

```diff
@@ -95,26 +70,50 @@ def update(index):
                 anchor="nw",
             )
         else:
-            statusbar_action.config(text=f"{random.choice(live_message)}")
-            mine_count = self.mines_around_square(coordinate)
-            if mine_count == 0:
-                self.open_squares(x - 1, y - 1)
-                self.open_squares(x - 1, y)
-                self.open_squares(x - 1, y + 1)
-                self.open_squares(x, y - 1)
-                self.open_squares(x, y + 1)
-                self.open_squares(x + 1, y - 1)
-                self.open_squares(x + 1, y)
-                self.open_squares(x + 1, y + 1)
-
-            if mine_count > 0:
-                canvas.create_text(
-                    coordinate[0] * button_size + (button_size / 2),
-                    coordinate[1] * button_size + (button_size / 2),
-                    text=str(mine_count),
-                    font=("helvetica", 17, "bold"),
-                    fill=color_chart[mine_count],
-                )
+            if (count_already_open + count_mine_locations == self.width * self.height):
+                self.game_status = GameStatus.game_won
+                statusbar_action.config(text=f"{random.choice(win_message)}")
+                frames = [
+                    PhotoImage(
+                        file=where_this_file_is / "doomguy.gif",
+                        format="gif -index %i" % (i),
+                    )
+                    for i in range(8)
+                ]
+                gif_label.place(relx=0.5, rely=0.5, anchor="center")
+
+                def update(index):
+                    frame = frames[index]
+                    index += 1
+                    if index >= 8:
+                        index = 0
+                    gif_label.configure(image=frame)
+                    if self is current_game:
+                        # New game not started yet, can keep animating
+                        root.after(100, update, index)
+
+                root.after(0, update, 0)
+            else:
+                statusbar_action.config(text=f"{random.choice(live_message)}")
+                mine_count = self.mines_around_square(coordinate)
+                if mine_count == 0:
+                    self.open_squares(x - 1, y - 1)
+                    self.open_squares(x - 1, y)
+                    self.open_squares(x - 1, y + 1)
+                    self.open_squares(x, y - 1)
+                    self.open_squares(x, y + 1)
+                    self.open_squares(x + 1, y - 1)
+                    self.open_squares(x + 1, y)
+                    self.open_squares(x + 1, y + 1)
+
+                if mine_count > 0:
```

Comment:
This is currently indented so that it doesn't run on the last click that wins the game. But it should run for the last click that wins the game.

Previously, the auto-opening code above also ran when you win the game, but it doesn't matter whether it runs or not.